core/runtime/src/main/java/io/quarkus/runtime/configuration/RuntimeConfigBuilder.java

```diff
@@ -12,7 +15,38 @@ public class RuntimeConfigBuilder implements SmallRyeConfigBuilderCustomizer {
     @Override
     public void configBuilder(final SmallRyeConfigBuilder builder) {
         new QuarkusConfigBuilderCustomizer().configBuilder(builder);
-        builder.withDefaultValue("quarkus.uuid", UUID.randomUUID().toString());
+        builder.withSources(new ConfigSource() {
```

Comment:
Looks interesting. If the `ConfigSource` are kept around at runtime, we might want to move this to a static nested class to avoid keeping the builder reference around?

That being said, better to wait for @radcortez 's feedback before polishing.

Reply:
If we want a quick solution, I'm okay with moving forward with this.

My idea was to add some sort of lazy type that can be declared in the mapping. This would allow keeping the declaration for documentation purposes and flattening sources if there are other cases.

Alternatively, we could just do this as a build step for the extensions that need it (it would have worked if it was done that way when it was first introduced). Unfortunately, we can't tell if users are now relying on it outside of the required extensions.

Reply:
> If we want a quick solution, I'm okay with moving forward with this.

That's totally up to you. The reason I vote for having this now is that we usually do the performance optimization in sprints (I don't mean the Agile sense of the word) and if we don't get this in now, who knows when we'll get around to it).

Reply:
In any case, I updated the PR to use an inner class instead of an anonymous one.

Reply:
I think it's worth merging this but I also agree having a lazy type in SmallRye Config would be nice for the future and we could revisit this patch then.

Reply:
BTW, another alternative (breaking) is to totally drop this from the Config system and add a specific API for it (this is similar to the issue we have with ports).